### PR TITLE
Update runner.py to shutdown VMs before dom0 tests and log OS info

### PR DESCRIPTION
**Pull request by zenmonkeykstop**
Adds two methods to QubesCI:

- systemInfo() - intended to be run at the start of a job, currently just dumps OS version info
- shutdown_sd_vms() - shuts down all SDW vms, run before `make test` to ensure VMs under test have latest updates from their TemplateVMs.

### CODE REVIEW

File: dom0/runner.py [hunk 1]
```diff
@@ -75,6 +75,21 @@ def __init__(self):
...
+        """
+        Shut down sd-workstation-tagged VMs.
+
+        This should be done before dom0 tests, to ensure that the
+        AppVMs tested have the latest TemplateVM changes.
+        """
+        self.logging.info("Shutting down SecureDrop Workstation VMs")
+        sdw_vms = [vm for vm in self.q.domains if "sd-workstation" in vm.tags]
+
+        for vm in sdw_vms:
```
Comment (legoktm) [inline dom0/runner.py:88]: Just noting that qvm-shutdown (which is Python importable, see https://github.com/QubesOS/qubes-core-admin-client/blob/main/qubesadmin/tools/qvm_shutdown.py) has a graceful loop that shuts the VMs down in order so you can avoid using --force. Don't think there's anything necessarily wrong with this approach though.

**Review by mig5 — APPROVED** [review]
Looks good to me! 

> systemInfo() - intended to be run at the start of a job, currently just dumps OS version info

Thought I'd note that the log file itself also shows the Qubes OS version in its name - technically actually the name of the VM it ran on, e.g:

https://ws-ci-runner.securedrop.org/2024-04-02-180000611211-dd82553490eb5cb632c611212087f4e562df0383-Qubes_4.2_A-update_20240402043959.log.txt

'Qubes _4.2_A' here is the machine the CI ran against. But this change is a nicer visual indicator :)

**Comment by mig5** [comment]
> I'm not sure it's possible in the current architecture, but can we do a proper reboot here? Though it might be too slow what we don't really want that.

Are you asking for a reboot of the app VMs or of dom0?

**Comment by legoktm** [comment]
I was thinking of a dom0 reboot, which would best mimic what normal user interaction would be. This PR takes care of rebooting the necessary VMs. (well it shuts them down, the test run will bring them back up)

**Comment by mig5** [comment]
Got it. I don't think that will work in the current setup, as the `make clone; make dev; make test` steps are happening inside dom0 already. From the 'outside', the VMware bastion is waiting for the `run.py` script in dom0 to complete. We can't currently 'intercept' in between the two requests and reboot the dom0, then 'get back in' and run the `make test` step (and on top of that, ensure it is still logging to the same log file).

To support that, I think we'd have to split up the `run.py` into separate 'stages' that run separate things. I'll have a think on it, but I don't want to block this PR on that either, as I don't know when I'd be able to figure something out.

**Comment by legoktm** [comment]
Yeah that makes sense, just curious. Also I suspect doing a reboot would just slow it down even further. Thanks as always :)

File: dom0/runner.py (hunk 1 above)
Comment (zenmonkeykstop) [reply dom0/runner.py:88]: My understanding is that `--force` in this case isn't a forced immediate halt like qvm-kill but just bypasses VM dependencies and issues a regular ACPI shutdown signal (so you can shut down a NetVM that has qubes using it, for example).

**Pull request merged**